src/Lykke.blue.Service.ReferralLinks/Controllers/ReferralLinksController.cs

```diff
@@ -0,0 +1,480 @@
+
+            await LogInfo(request, ControllerContext, referralLink.ToJson());
+
+            return Created(uri: $"api/referralLinks/{referralLink.Id}", value: new RequestRefLinkResponse { RefLinkId = referralLink.Id, RefLinkUrl = referralLink.Url });
+        }
+
+
+        /// <summary>
+        /// Request invitation referral link.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        [HttpPost("invitationLinks")]
+        [SwaggerOperation("RequestInvitationReferralLink")]
+        [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType(typeof(RequestRefLinkResponse), (int)HttpStatusCode.Created)]
+        [ProducesResponseType(typeof(RequestRefLinkResponse), (int)HttpStatusCode.OK)]
+        public async Task<IActionResult> RequestInvitationReferralLink([FromBody] InvitationReferralLinkRequest request)
+        {
+            if (request == null)
+            {
+                return await LogAndReturnBadRequest("", ControllerContext, Phrases.InvalidRequest);
+            }
+
+            if (String.IsNullOrEmpty(request.SenderClientId))
+            {
+                return await LogAndReturnBadRequest(request, ControllerContext, Phrases.InvalidSenderClientId);
+            }
+
+            var invitationLinkAlreadyCreated = _referralLinksService.GetInvitationLinksForSenderId(request.SenderClientId).ToList();
+            if (invitationLinkAlreadyCreated.Count == 1)
+            {
+
+                await LogInfo(request, ControllerContext, $"Invitation link already exists for SenderClientId {request.SenderClientId}. RefLinkId {invitationLinkAlreadyCreated.First().Id}");
+                return Ok(new RequestRefLinkResponse { RefLinkUrl = invitationLinkAlreadyCreated.First().Url, RefLinkId = invitationLinkAlreadyCreated.First().Id });
+            }
+            if (invitationLinkAlreadyCreated.Count > 1)
+            {
+                return await LogAndReturnInternalServerError(request, ControllerContext, $"There are multiple invitation links created for client {request.SenderClientId}! ");
+            }
+
+            if (!_referralLinksService.InvitationLinkForSenderIdExists(request.SenderClientId))
+            {
+                var referralLink = await _referralLinksService.CreateInvitationLink(request);
+
+                await LogInfo(request, ControllerContext, referralLink.ToJson());
+
+                return Created(uri: $"api/referralLinks/{referralLink.Id}", value: new RequestRefLinkResponse { RefLinkUrl = referralLink.Url, RefLinkId = referralLink.Id });
+            }
+            return await LogAndReturnBadRequest(request, ControllerContext, $"Invitation link for client {request.SenderClientId} already exists. Please try again!");
```

Comment:
Seems like this code is unreachable.

Reply:
Cant see the code now but I think it is intended to be executed if line 245 returns true - i.e the extra  check for ref link existence returns true.

Reply:
Fixed